kubernetes/plugin.go

```diff
@@ -73,66 +77,77 @@ func pluginTableDefinitions(ctx context.Context, d *plugin.TableMapData) (map[st
+					ctx = context.WithValue(ctx, contextKey("VersionSchemaStatus"), version.Schema.OpenAPIV3Schema.Properties["status"])
+					if len(version.Schema.OpenAPIV3Schema.Description) > 0 {
+						ctx = context.WithValue(ctx, contextKey("VersionSchemaDescription"), strings.TrimSuffix(version.Schema.OpenAPIV3Schema.Description, ".")+".")
+					}
+				}
+			}
+		}
+
+		// add the tables in sneak case
+		re := regexp.MustCompile(`[-.]`)
+		if tables["kubernetes_"+crd.Spec.Names.Singular] == nil {
+			ctx = context.WithValue(ctx, contextKey("TableName"), "kubernetes_"+crd.Spec.Names.Singular)
+			tables["kubernetes_"+crd.Spec.Names.Singular] = tableKubernetesCustomResource(ctx)
+		} else {
+			ctx = context.WithValue(ctx, contextKey("TableName"), "kubernetes_"+crd.Spec.Names.Singular+"_"+re.ReplaceAllString(crd.Spec.Group, "_"))
+			tables["kubernetes_"+crd.Spec.Names.Singular+"_"+re.ReplaceAllString(crd.Spec.Group, "_")] = tableKubernetesCustomResource(ctx)
+		}
+	}
 
 	return tables, nil
 }
 
-// Uncomment once aggregator functionality available with dynamic tables
-
-// func listK8sDynamicCRDs(ctx context.Context, cn *connection.ConnectionCache, c *plugin.Connection) ([]v1.CustomResourceDefinition, error) {
-// 	clientset, err := GetNewClientCRDRaw(ctx, cn, c)
-// 	if err != nil {
-// 		plugin.Logger(ctx).Error("listK8sDynamicCRDs", "GetNewClientCRDRaw", err)
-// 		return nil, err
-// 	}
-
-// 	input := metav1.ListOptions{
-// 		Limit: 500,
-// 	}
-
-// 	crds := []v1.CustomResourceDefinition{}
-
-// 	pageLeft := true
-// 	for pageLeft {
-// 		response, err := clientset.ApiextensionsV1().CustomResourceDefinitions().List(ctx, input)
-// 		if err != nil {
-// 			// Handle err at the plugin load time if config is not setup properly
-// 			if strings.Contains(err.Error(), "/apis/apiextensions.k8s.io/v1/customresourcedefinitions?limit=500") {
-// 				return nil, nil
-// 			}
-// 			plugin.Logger(ctx).Error("listK8sDynamicCRDs", "list_err", err)
-// 			return nil, err
-// 		}
-
-// 		if response.GetContinue() != "" {
-// 			input.Continue = response.Continue
-// 		} else {
-// 			pageLeft = false
-// 		}
-
-// 		crds = append(crds, response.Items...)
-// 	}
-
-// 	return crds, nil
-// }
+func listK8sDynamicCRDs(ctx context.Context, cn *connection.ConnectionCache, c *plugin.Connection) ([]v1.CustomResourceDefinition, error) {
+	clientset, err := GetNewClientCRDRaw(ctx, cn, c)
+	if err != nil {
+		plugin.Logger(ctx).Error("listK8sDynamicCRDs", "GetNewClientCRDRaw", err)
+		return nil, err
+	}
+
+	input := metav1.ListOptions{
+		Limit:          500,
+		TimeoutSeconds: types.Int64(5),
+	}
+
+	crds := []v1.CustomResourceDefinition{}
+
+	pageLeft := true
+	for pageLeft {
+		response, err := clientset.ApiextensionsV1().CustomResourceDefinitions().List(ctx, input)
+		if err != nil {
+			// Handle err at the plugin load time if config is not setup properly
+			if strings.Contains(err.Error(), "/apis/apiextensions.k8s.io/v1/customresourcedefinitions?limit=500") {
```

Comment:
- [ ] Can we add a better comment describing the error that we are handling here?